lib/slack-client.js

```diff
@@ -28,3 +32,56 @@ SlackClient.prototype.getUserName = function(userId) {
 SlackClient.prototype.getMessageText = function(message) {
   return message.text;
 };
+
+SlackClient.prototype.getReactions = function(channel, timestamp) {
+  return makeApiCall(this, 'reactions.get',
+    { channel: channel, timestamp: timestamp });
+};
+
+function makeApiCall(that, method, params) {
+  return new Promise(function(resolve, reject) {
+    var paramsStr, req;
+
+    params.token = process.env.HUBOT_SLACK_TOKEN;
+    paramsStr = JSON.stringify(params);
+
+    req = http.request(getHttpOptions(that, method, paramsStr), function(res) {
+      handleResponse(res, resolve, reject);
+    });
+    req.on('error', function(err) {
+      reject(new Error('failed to make Slack API request: ' + err.message));
+    });
+    req.end(paramsStr);
+  });
```

Comment:
Not a blocker, but you may consider simplifying all of this by using a package, e.g. [request-promise](https://github.com/request/request-promise).